components/sensor/kmeteriso.rst

```diff
@@ -0,0 +1,49 @@
+M5Stack KMeterISO I2C K-Type probe temperature sensor
+=====================================================
+
+.. seo::
+    :description: Instructions for setting up KMeterISO temperature sensors
+    :image: kmeteriso.jpg
+    :keywords: BME280
+
+The ``kmeteriso`` sensor platform allows you to use your KMeterISO
+(`product <https://docs.m5stack.com/en/unit/KMeterISO%20Unit>`__,
+`M5Stack`_) K-Type thermocouple temperature sensor with ESPHome.
+The :ref:`I²C <i2c>` is required to be set up in your configuration
+for this sensor to work.
+
+.. figure:: ../../images/kmeteriso.jpg
+    :align: center
+    :width: 50.0%
+
+    M5Stack KMeterISO temperature sensor.
+
+.. _M5Stack: https://docs.m5stack.com/en/unit/KMeterISO%20Unit
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    sensor:
+    - platform: kmeteriso
+      temperature:
+        name: Temperature
+      internal_temperature:
+        name: Internal temperature
+        entity_category: diagnostic
```

Comment:
Sorry just noticed this one more.

If this is recommended to be diagnostic, then it should be set in the python code as diagnostic by default, and removed from here.

Reply:
Thanks.  How do I default this entity to diagnostic in the python code, such that the user can also choose to make it *not* diagnostic?  Thanks in advance.

Reply:
I think I got it.  Check latest commit here as well as https://github.com/esphome/esphome/pull/5170/commits/f3af7f909a68c10cf308286f2d77d57425e868fb